packages/test/test-end-to-end-tests/src/test/SummarizeFetchValidation.spec.ts

```diff
@@ -0,0 +1,314 @@
+        mainDataStore._root.set("dataStore4", dataStore4.handle);
+        mainDataStore._root.set("dataStore5", dataStore5.handle);
+        mainDataStore._root.set("dataStore6", dataStore6.handle);
+
+        versionWrap = await incrementCellValueAndRunSummary(summarizer1, 3 /* expectedMatrixCellValue */);
+        assert(versionWrap.fetchCount === 0, "No fetch should have happened");
+        await provider.ensureSynchronized();
+        summarizer1.close();
+    });
+
+    it("Second summarizer from latest should not fetch", async () => {
+        const summarizer1 = await createSummarizer(provider, mainContainer);
+
+        let versionWrap = await incrementCellValueAndRunSummary(summarizer1, 1 /* expectedMatrixCellValue */);
+        assert(versionWrap.fetchCount === 0, "No fetch should have happened");
+        versionWrap = await incrementCellValueAndRunSummary(summarizer1, 2 /* expectedMatrixCellValue */);
+        assert(versionWrap.fetchCount === 0, "No fetch should have happened");
+        await provider.ensureSynchronized();
+        summarizer1.close();
+
+        const value = getAndIncrementCellValue(mainDataStore.matrix, 0, 0, "1");
+        assert(value === 3, "Value matches expected");
+
+        const summarizer2 = await createSummarizer(provider, mainContainer);
+        versionWrap = await incrementCellValueAndRunSummary(summarizer2, 4 /* expectedMatrixCellValue */);
+        assert(versionWrap.fetchCount === 0, "There should be one fetch after second summary");
+
+        versionWrap = await incrementCellValueAndRunSummary(summarizer2, 5 /* expectedMatrixCellValue */);
+        assert(versionWrap.fetchCount === 0, "No fetch should have happened");
+        summarizer2.close();
+    });
+
+    it("Loading Summary from older version should fetch", async () => {
+        const summarizerClient = await createSummarizer(provider, mainContainer);
+        let versionWrap = await incrementCellValueAndRunSummary(summarizerClient, 1 /* expectedMatrixCellValue */);
+        const summaryVersion = versionWrap.summaryVersion;
+        assert(versionWrap.fetchCount === 0, "No fetch should have happened");
+        assert(summaryVersion, "Summary version should be defined");
+        summarizerClient.close();
+
+        const secondSummarizer = await createSummarizer(provider, mainContainer);
+        versionWrap = await incrementCellValueAndRunSummary(secondSummarizer, 2 /* expectedMatrixCellValue */);
+        assert(versionWrap.fetchCount === 0, "No fetch should have happened");
+        versionWrap = await incrementCellValueAndRunSummary(secondSummarizer, 3 /* expectedMatrixCellValue */);
+        assert(versionWrap.fetchCount === 0, "No fetch should have happened");
+        await provider.ensureSynchronized();
+        secondSummarizer.close();
+
+        // Load summarizer from previous version triggers fetch.
+         const newContainer = await loadContainer(summaryVersion);
```

Comment:
Why create a container as well as summarizer? Why not just the summarizer?

Reply:
No particular reason - trying to make it a little different.